### PR TITLE
linuxPackages_5_13.perf: fix build

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
Also add optional libcap and zstd

The scripts directory was majorly changed, and  most than half the files needs to be patched to be usable.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
```
https://github.com/NixOS/nixpkgs/pull/128537

5 packages marked as broken and skipped:
linuxPackages_4_14.perf linuxPackages_4_19.perf linuxPackages_4_4.perf linuxPackages_4_9.perf linuxPackages_hardkernel_4_14.perf

1 package failed to build:
linuxPackages_xanmod.perf

10 packages built:
cargo-flamegraph linuxPackages-libre.perf linuxPackages.perf linuxPackages_5_12.perf linuxPackages_5_13.perf linuxPackages_5_4.perf linuxPackages_latest-libre.perf linuxPackages_lqx.perf linuxPackages_testing_bcachefs.perf linuxPackages_zen.perf
```

File: pkgs/os-specific/linux/kernel/perf.nix [hunk 1]
```diff
@@ -54,8 +60,10 @@ stdenv.mkDerivation {
     "-Wno-error=stringop-truncation"
   ];
 
-  postPatch = ''
+  postPatch = if (kernel.kernelOlder "5.13") then ''
     patchShebangs scripts/bpf_helpers_doc.py
+  '' else ''
+    patchShebangs scripts
```
Comment (Mic92) [inline pkgs/os-specific/linux/kernel/perf.nix:66]: Any harm in doing this?

```suggestion
  postPatch = ''
    patchShebangs scripts
  '';
```

Comment (jonringer) [reply pkgs/os-specific/linux/kernel/perf.nix:66]: probably not

Comment (jonringer) [reply pkgs/os-specific/linux/kernel/perf.nix]: applied

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128537` at b0b15156 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/64f6850c5049436e9be4389fc4ce59f8">linuxPackages_4_14.perf</a></li>
    <li><a href="https://gist.github.com/1710ffffa788380ec08186006496f0f5">linuxPackages_4_19.perf</a></li>
    <li><a href="https://gist.github.com/19bac1cb490489d007b9a491ef3e94dd">linuxPackages_4_4.perf</a></li>
    <li><a href="https://gist.github.com/8cd19799322b3207a4736f5694364697">linuxPackages_4_9.perf</a></li>
    <li><a href="https://gist.github.com/1ad0ad6a7e0c5e140ffdb1eea251b8a1">linuxPackages_hardkernel_4_14.perf</a> (linuxPackages_hardkernel_latest.perf)</li>
    <li><a href="https://gist.github.com/9ac8eab3220c30c222c7d1e908d05b27">linuxPackages_xanmod.perf</a></li>
  </ul>
</details>
<details>
  <summary>10 packages built successfully:</summary>
  <ul>
    <li>cargo-flamegraph</li>
    <li>linuxPackages-libre.perf</li>
    <li>linuxPackages.perf (linuxPackages_5_10.perf ,linuxPackages_hardened.perf)</li>
    <li>linuxPackages_5_12.perf (linuxPackages_latest.perf ,linuxPackages_latest_hardened.perf)</li>
    <li>linuxPackages_5_13.perf</li>
    <li>linuxPackages_5_4.perf</li>
    <li>linuxPackages_latest-libre.perf</li>
    <li>linuxPackages_lqx.perf</li>
    <li>linuxPackages_testing_bcachefs.perf</li>
    <li>linuxPackages_zen.perf</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by jonringer** [comment]
linuxPackages_4*.perf were already marked as broken. This just fixes the latest.

File: pkgs/os-specific/linux/kernel/perf.nix [hunk 2]
```diff
@@ -1,7 +1,11 @@
 { lib, stdenv, kernel, elfutils, python2, python3, perl, newt, slang, asciidoc, xmlto, makeWrapper
 , docbook_xsl, docbook_xml_dtd_45, libxslt, flex, bison, pkg-config, libunwind, binutils
 , libiberty, audit, libbfd, libopcodes, openssl, systemtap, numactl
-, zlib, withGtk ? false, gtk2 ? null
+, zlib
+, withGtk ? false
+, gtk2 ? null
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/kernel/perf.nix:6]:
```suggestion
, gtk2
```

File: pkgs/os-specific/linux/kernel/perf.nix [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
+, withZstd ? true, zstd
+, withLibcap ? true, libcap
 }:
 
 with lib;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/kernel/perf.nix:11]: and remove ``assert withGtk -> gtk2 != null;``

**Comment by Mic92** [comment]
Merge conflict.

**Pull request merged**